packages/core/react/src/useConnection.tsx

```diff
@@ -6,7 +6,13 @@ export interface ConnectionContextState {
 }
 
 export const ConnectionContext = createContext<ConnectionContextState>({} as ConnectionContextState);
+export const ConnectionDispatchContext = createContext<React.Dispatch<Connection>>({} as React.Dispatch<Connection>);
 
 export function useConnection(): ConnectionContextState {
     return useContext(ConnectionContext);
 }
+
+export function useSetConnection(): (connection: Connection) => void {
+    const dispatch = useContext(ConnectionDispatchContext);
+    return (connection: Connection) => dispatch(connection);
+}
```

Comment:
This seems overly complicated. Why not just `const [connection, setConnection] = useState(() => new Connection(...))` in the provider and then pass `setConnection` into the context?